### PR TITLE
build: to migrate from circleci to github workflows

### PR DESCRIPTION
**Pull request by yaswanth-deriv**
Changes:

I removed circleci and replaced with github workflows

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [x] Refactor code
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [X] Release

### CODE REVIEW

**Comment by ali-hosseini-deriv** [comment]
As you can see in the PR checks and circleci config we also have a test job for PRs.
you need to implement that one too.

File: .github/workflows/release_staging.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      uses: actions/checkout@v4
+    - name: Tag
+      uses: "./.github/actions/tag"
+    - name: Npm Install
+      uses: "./.github/actions/npm_install"
+    - name: Build
+      uses: "./.github/actions/build"
+      with:
+        NODE_ENV: staging
+    - name: Versioning
```
[the remark this answers is not in the document]
Comment (ali-hosseini-deriv) [reply .github/workflows/release_staging.yml:23]: you've missed this one @yaswanth-deriv

**Review by ali-hosseini-deriv — CHANGES_REQUESTED** [review]
please fix the EOF issues

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+name: Sinbad Test Workflow
+on:
+  pull_request:
+    branches:
+    - master
+jobs:
+  build_and_test:
+    name: Build And Test
```
Comment (niloofar-deriv) [inline .github/workflows/test.yml:8]:
```suggestion
    name: Build and Test
```

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    runs-on: ubuntu_latest
+    environment: Development
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+    - name: Npm Install
```
Comment (niloofar-deriv) [inline .github/workflows/test.yml:14]:
```suggestion
    - name: Install npm packages
```

File: .github/workflows/release_staging.yml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  build_test_and_publish:
+    name: Build, Test and Publish to Cloudflare Staging
+    runs-on: ubuntu-latest
+    environment: Staging
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+    - name: Tag
+      uses: "./.github/actions/tag"
+    - name: Npm Install
```
Comment (niloofar-deriv) [inline .github/workflows/release_staging.yml:17]:
```suggestion
    - name: Install npm packages
```

File: .github/actions/build/action.yml [hunk 5]
```diff
@@ -0,0 +1,7 @@
+name: build
```
Comment (niloofar-deriv) [inline .github/actions/build/action.yml:1]:
```suggestion
name: Build
```

File: .github/workflows/release_production.yml [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+  build_test_and_publish:
+    name: Build, Test and Publish to Cloudflare Pages Production
+    environment: Production
+    runs-on: ubuntu-latest
+    outputs:
+      RELEASE_VERSION: ${{ steps.extract_version.outputs.RELEASE_VERSION }}
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+    - name: NPM Install
```
Comment (niloofar-deriv) [inline .github/workflows/release_production.yml:16]:
```suggestion
    - name: Install npm packages
```

**Review by niloofar-deriv — COMMENTED** [review]
Nit

**Comment by niloofar-deriv** [comment]
Please ask @prince-deriv to approve your workflow 🙏

**Review by niloofar-deriv — APPROVED** [review]

File: .github/workflows/test.yml [hunk 7]
```diff
@@ -0,0 +1,17 @@
+name: Sinbad Test Workflow
+on:
+  pull_request:
+    branches:
+    - master
+jobs:
+  build_and_test:
+    name: Build and Test
+    runs-on: ubuntu_latest
+    environment: Development
```
Comment (ali-hosseini-deriv) [inline .github/workflows/test.yml:10]: we dont need any env vars or secrets here. 
so you can remove this.

**Review by ali-hosseini-deriv — CHANGES_REQUESTED** [review]
as @niloofar-deriv suggested we can setup node_modules caching for these workflows too.
please check other projects like deriv-static-content or deriv-api-docs for the implementation.

**Comment by ali-hosseini-deriv** [comment]
please remove all the environments: Production or Staging or Development.
they values are the same, so having different environments are just over engineering and doesn't add any value.

File: .github/workflows/test.yml (hunk 7 above)
Comment (yaswanth-deriv) [reply .github/workflows/test.yml:10]: OKay I will check it

**Review by ali-hosseini-deriv — APPROVED** [review]

**Review by habib-deriv — APPROVED** [review]

File: .github/workflows/test.yml [hunk 8]
```diff
@@ -0,0 +1,16 @@
+name: Sinbad Test Workflow
+on:
+  pull_request:
+    branches:
+    - master
+jobs:
+  build_and_test:
+    name: Build and Test
+    runs-on: ubuntu_latest
```
Comment (ali-hosseini-deriv) [inline .github/workflows/test.yml:9]: @yaswanth-deriv this has to be `ubuntu-latest` 
please fix it.
```suggestion
    runs-on: ubuntu-latest
```

**Review by ali-hosseini-deriv — APPROVED** [review]

**Pull request merged**